### PR TITLE
IGNITE-14432 Implement connection context managers for clients.

### PR DESCRIPTION
**Pull request by ivandasch**

### CODE REVIEW

File: tests/affinity/test_connection_context_manager.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+from pyignite import Client, AioClient
+
+
+@pytest.fixture
+def connection_param():
+    return [('127.0.0.1', 10800 + i) for i in range(1, 4)]
+
+
+@pytest.mark.parametrize('partition_aware', ['with_partition_aware', 'wo_partition_aware'])
```
Comment (isapego) [inline tests/affinity/test_connection_context_manager.py:26]: Why can not we just use `True`/`False` here?

**Review by isapego — APPROVED** [review]
Overall, looks good. Find my minor comment

File: tests/affinity/test_connection_context_manager.py (hunk 1 above)
Comment (ivandasch) [reply tests/affinity/test_connection_context_manager.py:26]: image
Hi. Pytest generates messages with params, it's easier to debug if some problems occurs.

Comment (isapego) [reply tests/affinity/test_connection_context_manager.py:26]: OK, I can see your point. Looks nice.

**Pull request closed without merging**